cli/commands/find/action.go

```diff
@@ -19,25 +20,101 @@ func Run(ctx context.Context, opts *Options) error {
 		d = d.WithHidden()
 	}
 
-	configs, err := d.Discover()
+	if opts.Dependencies || opts.External || opts.Sort == SortDAG {
+		d = d.WithDiscoverDependencies()
+	}
+
+	if opts.External {
+		d = d.WithDiscoverExternalDependencies()
+	}
+
+	cfgs, err := d.Discover(ctx, opts.TerragruntOptions)
+	if err != nil {
+		return errors.New(err)
+	}
+
+	switch opts.Sort {
+	case SortAlpha:
+		cfgs = cfgs.Sort()
+	case SortDAG:
+		q := queue.NewQueue(cfgs)
+		cfgs = q.Entries()
+	}
+
+	foundCfgs, err := discoveredToFound(cfgs, opts)
 	if err != nil {
 		return errors.New(err)
 	}
 
 	switch opts.Format {
-	case "text":
-		return outputText(opts, configs)
-	case "json":
-		return outputJSON(opts, configs)
+	case FormatText:
+		return outputText(opts, foundCfgs)
+	case FormatJSON:
+		return outputJSON(opts, foundCfgs)
 	default:
 		// This should never happen, because of validation in the command.
 		// If it happens, we want to throw so we can fix the validation.
 		return errors.New("invalid format: " + opts.Format)
 	}
 }
 
+type FoundConfigs []*FoundConfig
+
+type FoundConfig struct {
+	Type discovery.ConfigType `json:"type"`
+	Path string               `json:"path"`
+
+	Dependencies []string `json:"dependencies,omitempty"`
+}
+
+func discoveredToFound(configs discovery.DiscoveredConfigs, opts *Options) (FoundConfigs, error) {
+	foundCfgs := make(FoundConfigs, 0, len(configs))
+	errs := []error{}
+
+	for _, config := range configs {
+		if config.External && !opts.External {
+			continue
+		}
+
+		relPath, err := filepath.Rel(opts.WorkingDir, config.Path)
+		if err != nil {
+			errs = append(errs, errors.New(err))
+
+			continue
+		}
+
+		foundCfg := &FoundConfig{
+			Type: config.Type,
+			Path: relPath,
+		}
+
+		if !opts.Dependencies {
+			foundCfgs = append(foundCfgs, foundCfg)
+
+			continue
+		}
+
+		foundCfg.Dependencies = make([]string, len(config.Dependencies))
```

Comment:
Does it make sense to initialize `foundCfg.Dependencies` if `config.Dependencies` is empty?